bitcoin/src/util/psbt/mod.rs

```diff
@@ -437,6 +437,28 @@ impl PartiallySignedTransaction {
         // because there has been a new softfork that we do not yet support.
         Err(SignError::UnknownOutputType)
     }
+
+    /// Calculates transaction fee.
+    ///
+    /// 'Fee' being the amount that will be paid for mining a transaction with the current inputs
+    /// and outputs i.e., the difference in value of the total inputs and the total outputs.
+    ///
+    /// ## Errors
+    ///
+    /// - [`Error::MissingUtxo`] when UTXO information for any input is not present or is invalid.
+    /// - [`Error::NegativeFee`] if calculated value is negative.
+    /// - [`Error::FeeOverflow`] if an integer overflow occurs.
+    pub fn fee(&self) -> Result<u64, Error> {
```

Comment:
Oh, wait, sorry for being late I only realized now. We should return `Amount` here.

Reply:
done, I left calculations using `u64` because psbt values are `u64` (why btw?)

Reply:
Yep, that's correct.

> why

An oversight in API design. We have an issue to change it but nobody did it yet. (We have a bunch of APIs we want to improve so if you're surprised next time it's probably the same reason.)